### DISCUSSION TITLE
Is there a withCookie() method for Livewire tests?

### DISCUSSION THREAD

**Discussion by nagra** [livewire/livewire#2895] [1. Help]
Hi,

I am struggling to find a way to set a cookie to assert some behaviour in a Livewire test.
I was hoping there was a method similar to `withCookie` for normal Laravel tests, but could only find `Livewire::withQueryParams` in the Livewire documentation.

I've tried the below code, but it doesn't work.

```
public function test_cookie()
{
    $this->actingAs($user = User::factory()->create());
    
    Cookie::queue('username', 'testName');

    Livewire::test(CreateSquad::class)
        ->assertSet('squadUsername', 'testName');
}
```  

The underlying class retrieves the cookie as follows:
`request()->cookie('try-username');`

Any help is much appreciated 🙏.
Thanks.

**Comment by edalzell** [comment]
There will be when https://github.com/livewire/livewire/pull/6822 gets merged
